packages/core/src/renderer/components/+namespaces/namespace-tree-view.tsx

```diff
@@ -0,0 +1,144 @@
+import namespaceStoreInjectable from "./store.injectable";
+import { SubnamespaceBadge } from "./subnamespace-badge";
+
+interface NamespaceTreeViewProps {
+  root: Namespace;
+}
+
+interface Dependencies {
+  namespaceStore: NamespaceStore;
+  getDetailsUrl: GetDetailsUrl;
+}
+
+function isNamespaceControlledByHNC(namespace: Namespace) {
+  const hierarchicalNamesaceControllerLabel = "hnc.x-k8s.io/included-namespace=true";
+
+  return namespace.getLabels().find(label => label === hierarchicalNamesaceControllerLabel);
+}
+
+function NonInjectableNamespaceTreeView({ root, namespaceStore, getDetailsUrl }: Dependencies & NamespaceTreeViewProps) {
+  const hierarchicalNamespaces = namespaceStore.getByLabel(["hnc.x-k8s.io/included-namespace=true"]);
+  const [expandedItems, setExpandedItems] = React.useState<string[]>(hierarchicalNamespaces.map(ns => `namespace-${ns.getId()}`));
+
+  function renderChildren(parent: Namespace) {
+    const children = hierarchicalNamespaces.filter(ns =>
+      ns.getLabels().find(label => label === `${parent.getName()}.tree.hnc.x-k8s.io/depth=1`),
+    );
+
+    return children.map(child => (
+      <StyledTreeItem
+        key={`namespace-${child.getId()}`}
+        nodeId={`namespace-${child.getId()}`}
+        data-testid={`namespace-${child.getId()}`}
+        onIconClick={(evt) =>{
+          toggleNode(`namespace-${child.getId()}`);
+          evt.stopPropagation();
+        }}
+        label={(
+          <>
+            <Link key={child.getId()} to={getDetailsUrl(child.selfLink)}>
+              {child.getName()}
+            </Link>
+            {" "}
+            <SubnamespaceBadge
+              id={`namespace-details-badge-for-${child.getId()}`}
+              namespace={child}
+            />
+          </>
+        )}
+      >
+        {renderChildren(child)}
+      </StyledTreeItem>
+    ));
+  }
+
+  function toggleNode(id: string) {
+    if (expandedItems.includes(id)) {
+      setExpandedItems(expandedItems.filter(item => item !== id));
+    } else {
+      setExpandedItems([...expandedItems, id]);
+    }
+  }
+
+  if (!isNamespaceControlledByHNC(root)) {
+    return null;
+  }
+
+  return (
+    <div data-testid="namespace-tree-view">
+      <DrawerTitle>Tree View</DrawerTitle>
+      <TreeView
+        defaultExpanded={[`namespace-${root.getId()}`]}
+        defaultCollapseIcon={<MinusSquare />}
+        defaultExpandIcon={<PlusSquare />}
+        defaultEndIcon={(<div style={{ opacity: 0.3 }}><MinusSquare /></div>)}
+        expanded={expandedItems}
+      >
+        <StyledTreeItem
+          nodeId={`namespace-${root.getId()}`}
+          label={root.getName()}
+          data-testid={`namespace-${root.getId()}`}
+          onIconClick={(evt) => {
+            toggleNode(`namespace-${root.getId()}`);
+            evt.stopPropagation();
+          }}
+        >
+          {renderChildren(root)}
+        </StyledTreeItem>
+      </TreeView>
+    </div>
+  );
+}
+
+function MinusSquare() {
```

Comment:
_nit:_ 

```suggestion
function MinusSquareIcon() {
```

_btw,_ good idea to add custom inline svg icons like this.